File: packages/ios/Sources/Components/RouteSummary/RouteSummary.figma.swift
import SwiftUI
import Figma

struct KozmosRouteSummaryConnect: FigmaConnect {
    let component = KozmosRouteSummary<Image>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8296"

    @FigmaString("ETA Text")
    var etaText: String = "6 min"

    @FigmaString("Distance Text")
    var distanceText: String = "320 m"

    @FigmaEnum(
        "State",
        mapping: [
            "Preview": KozmosRouteSummaryState.preview,
            "Active": KozmosRouteSummaryState.active
        ]
    )
    var state: KozmosRouteSummaryState = .preview

    var body: some View {
        KozmosRouteSummary(
            etaText: self.etaText,
            distanceText: self.distanceText,
            state: self.state,
            onEndRoute: {},
            onStartNavigation: {}
        ) {
            Image(systemName: "figure.walk")
        }
    }
}

File: packages/android/src/main/java/com/kozmos/components/RouteSummary/RouteSummary.figma.kt
package com.kozmos.components.routesummary

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8296")
class KozmosRouteSummaryConnect {
    @FigmaProperty(FigmaType.Text, "ETA Text")
    val etaText: String = "6 min"

    @FigmaProperty(FigmaType.Text, "Distance Text")
    val distanceText: String = "320 m"

    @FigmaProperty(FigmaType.Enum, "State")
    val state: KozmosRouteSummaryState = Figma.mapping(
        "Preview" to KozmosRouteSummaryState.Preview,
        "Active" to KozmosRouteSummaryState.Active
    )

    @Composable
    fun ComponentExample() {
        KozmosRouteSummary(
            etaText = etaText,
            distanceText = distanceText,
            onEndRoute = {},
            state = state,
            onStartNavigation = {}
        )
    }
}
